### PR TITLE
Remove sdk.MustSortJSON from channel types Acknowledgement

### PR DESCRIPTION
**Pull request by expertdicer**
## Description

closes: #3920

### Commit Message / Changelog Entry

```text
chore: remove sdk.MustSortJSON call when marshalling Acknowledgement
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/types/acknowledgement.go [hunk 1]
```diff
@@ -70,5 +69,9 @@ func (ack Acknowledgement) Success() bool {
 // Acknowledgement implements the Acknowledgement interface. It returns the
 // acknowledgement serialised using JSON.
 func (ack Acknowledgement) Acknowledgement() []byte {
-	return sdk.MustSortJSON(SubModuleCdc.MustMarshalJSON(&ack))
+	res, err := json.Marshal(SubModuleCdc.MustMarshalJSON(&ack))
```
Comment (colin-axner) [inline modules/core/04-channel/types/acknowledgement.go:72]:
```suggestion
	res, err := json.Marshal(&ack)
```
I think we switch to just marshaling the acknowledgement?

Comment (expertdicer) [reply modules/core/04-channel/types/acknowledgement.go]: Yes. I will just remove the `MustSortJSON` and leave the `MustMarshalJSON` untouch

**Review by chatton — APPROVED** [review]

**Review by crodriguezvega — APPROVED** [review]

**Comment by crodriguezvega** [comment]
@expertdicer can you rebase with main? It would actually also be very useful if you could give us write permissions to your fork, so that we can push updates to your PRs. Thanks!

**Comment by expertdicer** [comment]
> @expertdicer can you rebase with main? It would actually also be very useful if you could give us write permissions to your fork, so that we can push updates to your PRs. Thanks!

I have already invited you guys to my fork. You might need to accept the invitation.

**Comment by crodriguezvega** [comment]
> I have already invited you guys to my fork. You might need to accept the invitation. TYSM

Ouch, sorry I had missed it.

**Pull request merged**